src/models/models.ts

```diff
@@ -0,0 +1,52 @@
+export enum Status {
+  Available = 'available for checkout',
+  InStorage = 'in storage',
+  NeedsRepair = 'needs repair',
+}
+
+export interface LibraryItem {
+  readonly inventory_id: string;
+  readonly item_status: Status[];
+  readonly created_at: Date;
+  readonly updated_at: Date;
+  readonly name: string;
+  readonly quantity: number;
+  readonly image_url?: string | null;
+  readonly tags?: string[] | null;
+  readonly description?: string | null;
+  readonly accessories?: string[] | null;
+}
```

Comment:
Genuinely curious why `readonly` was added to all of these, do you have a reasoning for it?  Just asking because I've never used one in an interface, but only in Class implementations. 

I think it'd be safe to remove the `readonly` bits from all these interfaces unless ya'll have a specific reason to include them.

Reply:
Oh the readonly is for immutability, my understanding is that even tho it can be rare, it can prevent bugs from happing. I think it overall makes it harder to break the code when using these models.

Reply:
[It's just a bit of a dummy-blocker](https://www.typescriptlang.org/docs/handbook/2/objects.html#readonly-properties) - while the properties may be overwritten later, the extra check from `readonly` keeps us from overwriting any of these properties during development.

Reply:
I'm cool leaving them as `readonly` for now, I guess my concern was when we'd have to manipulate the hard coded value to send out POST requests, but maybe it's not a big deal and will work fine.

Reply:
so in that regard in the code, we would just have to make copies( with mutation ) of the entire object and send that instead in the POST request, or any other place we need to, instead.